GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -864,47 +886,24 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
 
                 FilterChanged?.Invoke(this, new FilterChangedEventArgs(_filterInfo));
 
-                _initialLoad = true;
-
                 _buildServerWatcher.CancelBuildStatusFetchOperation();
 
-                DisposeRevisionReader();
-
-                var newCurrentCheckout = Module.GetCurrentCheckout();
-                GitModule capturedModule = Module;
-
-                // If the current checkout changed, don't get the currently selected rows, select the
-                // new current checkout instead.
-                if (newCurrentCheckout == CurrentCheckout)
-                {
-                    _selectedObjectIds = _gridView.SelectedObjectIds;
-                }
-                else
-                {
-                    // This is a new checkout, so ensure the variable is cleared out.
-                    _selectedObjectIds = null;
-                }
-
-                CurrentCheckout = newCurrentCheckout;
                 base.Refresh();
 
                 IndexWatcher.Reset();
 
-                SelectInitialRevision(newCurrentCheckout);
-
                 _isReadingRevisions = true;
-                Subject<GitRevision> revisions = new();
+                _superprojectCurrentCheckout = null;
+                Subject<GitRevision> observeRevisions = new();
                 _revisionSubscription?.Dispose();
-                _revisionSubscription = revisions
+                _revisionSubscription = observeRevisions
                     .ObserveOn(ThreadPoolScheduler.Instance)
                     .Subscribe(OnRevisionRead, OnRevisionReaderError, OnRevisionReadCompleted);
 
-                _revisionReader ??= new RevisionReader();
-
-                // Find all ambiguous refs (including stash, notes etc)
-                IReadOnlyList<IGitRef> refs = (getRefs ?? Module.GetRefs)(RefsFilter.NoFilter);
-                _ambiguousRefs = GitRef.GetAmbiguousRefNames(refs);
+                System.Threading.CancellationToken cancellationToken = _refreshRevisionsSequence.Next();
 
+                _selectionTimer.Enabled = false;
+                _selectionTimer.Stop();
```

Comment:
```suggestion
                _selectionTimer.Stop();
```
These pairs have been in the existing code but they are synonymous.

Reply:
There is no point in letting the timer run before revisions may be added

Reply:
I will revert this, there are no revisions shown yet anyway, so the change can be in the early phase.

Reply:
Investigated this some more. Removing this.
This was likely an attempt to force run the selection timer after grid was loaded. Initially it seem to be used at the first load only but was running at every update.
However, as  git-log was async already before this PR (and has been for a long time), it was just forcing an update for no reason as the selection timer would trigger one anyway (maybe even before this timer fired).
It may have been a way to clear info in the tabs similar to what RussKie added in this PR for side panel, but it seem to OK without.
(Will play around some more.)

Reply:
Adding this, to initiate a refresh of tab info.
This is initiated before Refs are read, previously it could take some time before this occurred.
(Removing the change that refreshed only the Commit info, in tab or in the panel)